gdrive_sync/api.py

```diff
@@ -121,6 +121,80 @@ def query_files(query: str, fields: str) -> Iterable[Dict]:
             yield file_obj
 
 
+def _get_or_create_drive_file(
+    file_obj: Dict,
+    drive_path: str,
+    website: Website,
+    sync_date: Optional[datetime],
+    replace_file_with_the_same_name: bool = True,
+) -> Optional[DriveFile]:
+    """
+    Determines if `file_obj` is a new or updated file and returns a new or updated
+    DriveFile respectively.
+    Returns None if no change is detected.
+    """
+    existing_file = DriveFile.objects.filter(file_id=file_obj.get("id")).first()
+    if (
+        existing_file
+        and existing_file.checksum == file_obj.get("md5Checksum", "")
+        and existing_file.name == file_obj.get("name")
+        and existing_file.status
+        in (
+            DriveFileStatus.COMPLETE,
+            DriveFileStatus.UPLOADING,
+            DriveFileStatus.UPLOAD_COMPLETE,
+            DriveFileStatus.TRANSCODING,
+        )
+    ):
+        # For inexplicable reasons, sometimes Google Drive continuously updates
+        # the modifiedTime of files, so only update the DriveFile if the checksum or name changed,
+        # and the status indicates that the file processing is not complete or in progress.
+        return
+
+    file_data = {
+        "drive_path": drive_path,
+        "name": file_obj.get("name"),
+        "website": website,
+        "mime_type": file_obj.get("mimeType"),
+        "checksum": file_obj.get("md5Checksum"),
+        "modified_time": file_obj.get("modifiedTime"),
+        "created_time": file_obj.get("createdTime"),
+        "download_link": file_obj.get("webContentLink"),
+        "sync_error": None,
+        "sync_dt": sync_date,
+    }
+
+    if existing_file:
+        for k, v in file_data.items():
+            setattr(existing_file, k, v)
+        existing_file.save()
+        drive_file = existing_file
+    else:
+        existing_file_same_path = DriveFile.objects.filter(
```

Comment:
It might be good to come up with some different names to clarify what the difference is between `existing_file` and `existing_file_same_path`.  A couple ideas:

`existing_file_same_id` and `existing_file_same_path`
`existing_file` and `other_file_same_path`